File: haskell/6kyu.build-a-pile-of-cubes.hs
module Codewars.Kata.PileOfCubes where

findNb :: Integer -> Integer
findNb n
    | 4 * n == k^2 * (k+1)^2  = k
    | otherwise               = -1
  where k = floor $ sqrt $ sqrt $ fromInteger (4 * n)

File: javascript/6kyu.build-a-pile-of-cubes.js
const helper = (rest, n) => {
  switch (true) {
    case rest < 0:    return -1;
    case rest === 0:  return n -1;
    default:          return helper(rest - n*n*n, n + 1);
  }
};

const findNb = (n) => helper(n, 1);